sample_apps/simple_app_python/example_app.py

```diff
@@ -0,0 +1,318 @@
+    server_app_port         = parsed_args.server_app_port
+    print_all               = parsed_args.print_all
+
+    if print_all:
+        print('INITIALIZED_CERT_SIZE = ', policy_key.INITIALIZED_CERT_SIZE)
+        print('INITIALIZED_CERT[] size = ', len(policy_key.INITIALIZED_CERT))
+        print('pf_file_name = ', pf_file_name)
+
+    purpose = 'authentication'
+    enclave_type = 'simulated-enclave'
+    store_file = app1_data_dir + '/' + policy_store_file
+
+    cctm = cfm.cc_trust_manager(enclave_type, purpose, store_file)
+
+    if print_all:
+        print('Size of bytes() array =', len(bytes(policy_key.INITIALIZED_CERT)))
+
+    # -------------------------------------------------------------------------
+    # Convert policy_key certificate to a byte-stream
+    result = cctm.init_policy_key(bytes(policy_key.INITIALIZED_CERT))
+    if result is False:
+        print(fnl(), 'Cannot init policy key\n')
+        sys.exit(1)
+
+    # -------------------------------------------------------------------------
+    # Open hard-coded key / platform endorsement & app-measurement files
+    # and read-in data as a bytestream.
+    attest_key_file_name = app1_data_dir + '/' + attest_key_file
+    with open(attest_key_file_name, 'rb') as attest_key_file_fh:
+        attest_key_bin = attest_key_file_fh.read()
+
+    measurement_file_name = app1_data_dir + '/' + measurement
+    with open(measurement_file_name, 'rb') as measurement_fh:
+        example_app_measurement = measurement_fh.read()
+
+    attest_endorsement_file_name = app1_data_dir + '/' + pf_attest_endorsement
+    with open(attest_endorsement_file_name, 'rb') as attest_endorsement_fh:
+        platform_attest_endorsement_bin = attest_endorsement_fh.read()
+
+    if print_all:
+        print('attest_key_file_name         =', attest_key_file_name,
+              ', attest_key_bin len =', len(attest_key_bin))
+
+        print('measurement_file_name        =', measurement_file_name,
+              ', measurement len = ', len(example_app_measurement))
+
+        print('attest_endorsement_file_name =', attest_endorsement_file_name,
+              ', pf_attest_endorsement_bin =', len(platform_attest_endorsement_bin))
+        print(' ')
+
+    result = cctm.python_initialize_simulated_enclave(attest_key_bin,
+                                                      example_app_measurement,
+                                                      platform_attest_endorsement_bin)
+    if result is False:
+        print(fnl(), 'Cannot initialize enclave of type', enclave_type,
+                     'for purpose of', purpose, '\n')
+        sys.exit(1)
+
+    public_key_alg = "rsa-2048"
+    symmetric_key_alg = "aes-256-cbc-hmac-sha256"
+
+    # -------------------------------------------------------------------------
+    # Should succeed with valid key algorithm names, after policy key has been
+    # initialized
+    if operation == 'cold-init':
```

Comment:
@rgerganov -- Even while developing this program, I did consider modularizing inline code into sub-functions.

It seemed like needless refactoring. The body of each block is currently fairly small. To refactor them into functions, I'd have to deal with passing-down arguments and so on. Can be done; but just seemed a bit risky at this late stage. The way it's coded now, one benefit is that the scope of Python objects and CC stuff is very obvious as everything is in-line and at the same "code-block" level.

If you don't insist, I'll skip this rework, so that I can get this in (tomorrow) after you've given this PR a final look.

Reply:
Fair enough.